apps/web/pages/api/auth/reset-password.ts

```diff
@@ -50,6 +51,20 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
         emailVerified: new Date(),
       },
     });
+
+    // Change Identity Provider to CAL if it was previously Google
+    if (updatedUser.identityProvider === IdentityProvider.GOOGLE) {
```

Comment:
Why do we check for previous provider to be Google. Technically it can be anything right?

Reply:
We only have CAL, GOOGLE and SAML as Identity Provider

Reply:
@sean-brydon whats your take on this?

Reply:
We could have more providers later. Because we want the identityProvider to be CAL if password reset happens, regardless of what it was earlier, I don't see a reason for this check.

Reply:
Also, I think that just requesting a password reset will change the identity provider to CAL. Shouldn't we do it only through the confirm password reset flow?

In the current approach, anyone can get someone else's identity provider to be changed.

Reply:
> Also, I think that just requesting a password reset will change the identity provider to CAL. Shouldn't we do it only through the 'confirm password reset' flow?

> In the current approach, anyone can get someone else's identity provider to be changed.

no , the provider and password is only updated after entering new password from reset password link (in email). clicking on forgot password just send the email and doesn't update anything

Reply:
Cool